newrelic/data_source_newrelic_notifications_destination.go

```diff
@@ -67,10 +69,18 @@ func dataSourceNewRelicNotificationDestinationRead(ctx context.Context, d *schem
 	providerConfig := meta.(*ProviderConfig)
 	accountID := selectAccountID(providerConfig, d)
 	updatedContext := updateContextWithAccountID(ctx, accountID)
-	id := d.Get("id").(string)
-	filters := ai.AiNotificationsDestinationFilter{ID: id}
+	var filters ai.AiNotificationsDestinationFilter
 	sorter := notifications.AiNotificationsDestinationSorter{}
 
+	nameValue, nameOk := d.Get("name").(string)
+	if nameOk && nameValue != "" {
+		filters = ai.AiNotificationsDestinationFilter{Name: nameValue}
+	}
+	idValue, idOk := d.Get("id").(string)
+	if idOk && idValue != "" {
+		filters = ai.AiNotificationsDestinationFilter{ID: idValue}
+	}
+
```

Comment:
@lzaga-newrelic small question - the ID is added to the filter only if the name is not found and has not been added to the filter. Does this mean the name takes precedence over the ID?

Reply:
Never mind, sorry - I realised only one of **id** and **name** can be specified and upon testing this, found that the condition has been enforced and throws an error if both **id** and **name** are specified.